casper/src/test/scala/coop/rchain/casper/engine/Setup.scala

```diff
@@ -133,8 +136,12 @@ object Setup {
           estimateBlockHash: BlockHash
       ): Task[Float] = Task.pure(1.0f)
     }
-    implicit val lastFinalizedBlockCalculator = LastFinalizedBlockCalculator[Task](0f)
-    implicit val synchronyConstraintChecker   = SynchronyConstraintChecker[Task](0d)
+    val finalizedBlocksStore  = InMemoryKeyValueStore[Task]()
+    val finalizedBlocksStoreT = finalizedBlocksStore.toTypedStore(codecBlockHash, codecBlockHash)
```

Comment:
As you need to store only list of hashes, you can store only keys, so for value use codec `ignore(0)`, it is defined in scodec package. And use `KeyValueTypedStore[F, BlockHash, Unit]`

Reply:
Nice idea